sdk/azcore/runtime/policy_bearer_token.go

```diff
@@ -51,6 +51,14 @@ func acquire(state acquiringResourceState) (newResource exported.AccessToken, ne
 	return tk, tk.ExpiresOn, nil
 }
 
+// shouldRefresh determines whether the token should be refreshed. It's a variable so tests can replace it.
+var shouldRefresh = func(tk exported.AccessToken, _ acquiringResourceState) bool {
+	if tk.RefreshOn.IsZero() {
+		return tk.ExpiresOn.Add(-5 * time.Minute).Before(time.Now())
+	}
+	return tk.RefreshOn.Before(time.Now())
```

Comment:
Do we not need a window here, or is that baked into the value in `RefreshOn`?

Reply:
I'll add a comment. We take the value of RefreshOn literally because it defines the authority's suggested refresh window